scripts/dts/gen_defines.py

```diff
@@ -53,6 +54,28 @@ def main():
 
     active_compats = set()
 
+    zdid = 1
+    node2zdid = {}
+
+    # Assign a ZDID (Zephyr Device ID) to all nodes before we start
```

Comment:
Device id should be assigned by edts (not gen_defines.py). 

Device id seems to become a vital part of device handling with DTS in Zephyr. Having device id in edts makes it instantaneously available to any tool (e.g. code generation) that works on DTS in Zephyr. Otherwise the device id has to be extracted from the generated header file, which is just a waste of (maintenance) effort.